src/main/java/com/mongodb/kafka/connect/sink/converter/LazyBsonDocument.java

```diff
@@ -35,13 +35,13 @@
 public class LazyBsonDocument extends BsonDocument {
   private static final long serialVersionUID = 1L;
 
-  private transient SinkRecord record;
-  private transient Type dataType;
-  private transient BiFunction<Schema, Object, BsonDocument> converter;
+  private final transient SinkRecord record;
+  private final transient Type dataType;
+  private final transient BiFunction<Schema, Object, BsonDocument> converter;
 
   private BsonDocument unwrapped;
 
-  enum Type {
+  public enum Type {
```

Comment:
Does this need to be public?

Reply:
Yes for a test case.

Reply:
I'd prefer not to add to the API surface only in order to write a test.  Is there any alternative?

Reply:
The API surface is essentially private as users cannot be expected to use classes directly.

Reply:
Ah, that's a good point.